orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt

```diff
@@ -97,35 +98,54 @@ open class RunTaskHandler
         val exceptionDetails = exceptionHandlers.shouldRetry(e, taskModel.name)
         if (exceptionDetails?.shouldRetry ?: false) {
           log.warn("Error running ${message.taskType.simpleName} for ${message.executionType.simpleName}[${message.executionId}]")
-          queue.push(message, task.backoffPeriod())
-          trackResult(stage, task.javaClass, RUNNING)
+          queue.push(message, task.backoffPeriod(taskModel))
+          trackResult(stage, taskModel, RUNNING)
         } else if (e is TimeoutException && stage.getContext()["markSuccessfulOnTimeout"] == true) {
           queue.push(CompleteTask(message, SUCCEEDED))
         } else {
           log.error("Error running ${message.taskType.simpleName} for ${message.executionType.simpleName}[${message.executionId}]", e)
           stage.getContext()["exception"] = exceptionDetails
           repository.storeStage(stage)
           queue.push(CompleteTask(message, stage.failureStatus()))
-          trackResult(stage, task.javaClass, stage.failureStatus())
+          trackResult(stage, taskModel, stage.failureStatus())
         }
       }
     }
   }
 
-  private fun trackResult(stage: Stage<*>, taskType: Class<Task>, status: ExecutionStatus) {
+  private fun trackResult(stage: Stage<*>, taskModel: com.netflix.spinnaker.orca.pipeline.model.Task, status: ExecutionStatus) {
     val id = registry.createId("task.invocations")
       .withTag("status", status.toString())
       .withTag("executionType", stage.getExecution().javaClass.simpleName)
-      .withTag("stageType", stage.getType())
-      .withTag("taskType", taskType.simpleName)
       .withTag("isComplete", status.isComplete.toString())
-      .withTag("sourceApplication", stage.getExecution().getApplication())
+      .withTag("application", stage.getExecution().getApplication())
       .let { id ->
         stage.getContext()["cloudProvider"]?.let {
           id.withTag("cloudProvider", it.toString())
         } ?: id
       }
     registry.counter(id).increment()
+
+    registry.counter(
+      // a per duration bucket counter (5m, 15m, 30m, 60m)
+      registry.createId("task.invocations.${bucketDuration(taskModel)}").withTags(id.tags())
```

Comment:
The larger the tag space, the shorter duration that tags can be analyzed over ... unsure if have a tag with 6 possible values would cause us grief.

Maybe I'll flip it to a tag and we can run it for a bit.